### PR TITLE
Sdcb 9993 [Appium 2.0 Support]Java CS 

### PR DESCRIPTION
**Pull request by eryk-tkaczewski**

### CODE REVIEW

File: samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.android.clientside.properties [hunk 1]
```diff
@@ -1,8 +1,9 @@
 platformName=Android
-deviceName=Android Device
 fullReset=false
 noReset=true
+automationName=uiautomator2
 bitbar_target=Android
```
Comment (michal-przytarski) [inline samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.android.clientside.properties:5]: `bitbar_target` is deprecated and should be removed

File: samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.android.clientside.properties [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
 bitbar_project=Android Java Appium Client Side Example Project
```
Comment (michal-przytarski) [inline samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.android.clientside.properties:6]: `_` should be replaced with `:` if possible

File: samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.android.clientside.properties (hunk 1 above)
Comment (MaciejLendzion-smartbear) [reply samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.android.clientside.properties:3]: bump

File: samples/testing-frameworks/appium/client-side/java/src/test/java/com/testdroid/appium/ios/sample/IosAppiumExampleTest.java [hunk 3]
```diff
@@ -26,13 +26,13 @@ public void mainPageTest() {
         wd.manage().timeouts().implicitlyWait(60, TimeUnit.SECONDS);
         wd.findElement(By.xpath("//*[contains(@name, 'answer1')]")).click();
         WebElement element = wd.findElement(By.xpath("//*[contains(@name, 'userName')]"));
-        takeScreenshot("answer1_element_clicked");
+//        takeScreenshot("answer1_element_clicked");
```
Comment (MaciejLendzion-smartbear) [inline samples/testing-frameworks/appium/client-side/java/src/test/java/com/testdroid/appium/ios/sample/IosAppiumExampleTest.java:29]: Are we no longer taking screenshots in test or you left it like that for testing purposes?

File: samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.android.clientside.properties (hunk 1 above)
Comment (eryk-tkaczewski) [reply samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.android.clientside.properties]: Done

File: samples/testing-frameworks/appium/client-side/java/src/test/java/com/testdroid/appium/BaseTest.java [hunk 4]
```diff
@@ -70,18 +69,35 @@ private String getAppCapability() {
...
+            String strippedKey = key.substring("bitbar_".length());
+            bitbarOptions.setCapability(strippedKey, value);
+        } else {
             desiredCapabilities.setCapability(key, value);
         }
-        return desiredCapabilities;
     }
+    bitbarOptions.setCapability("apiKey", getApiKey());
+    desiredCapabilities.setCapability("appium:options", appiumOptions);
+    desiredCapabilities.setCapability("bitbar:options", bitbarOptions);
```
[the remark this answers is not in the document]
Comment (eryk-tkaczewski) [reply samples/testing-frameworks/appium/client-side/java/src/test/java/com/testdroid/appium/BaseTest.java]: Keeping it that way least interferes with the classes themselves, if we were to remove the .properties files we would have to rewrite all the sample logic, the way we run it and write a new README.

[the remark this answers is not in the document]
Comment (eryk-tkaczewski) [reply samples/testing-frameworks/appium/client-side/java/src/test/java/com/testdroid/appium/BaseTest.java]: imo it is not worth it

[the remark this answers is not in the document]
Comment (MaciejLendzion-smartbear) [reply samples/testing-frameworks/appium/client-side/java/src/test/java/com/testdroid/appium/BaseTest.java]: I mean it's still not the scope of this task to refactor all the sample so if you find it too complex then I'm ok with it.

File: samples/testing-frameworks/appium/client-side/java/src/test/java/com/testdroid/appium/ios/sample/IosAppiumExampleTest.java [hunk 5]
```diff
@@ -26,13 +26,14 @@ public void mainPageTest() {
         wd.manage().timeouts().implicitlyWait(60, TimeUnit.SECONDS);
         wd.findElement(By.xpath("//*[contains(@name, 'answer1')]")).click();
         WebElement element = wd.findElement(By.xpath("//*[contains(@name, 'userName')]"));
-//        takeScreenshot("answer1_element_clicked");
+        takeScreenshot("dupa");
```
[the remark this answers is not in the document]
Comment (eryk-tkaczewski) [reply samples/testing-frameworks/appium/client-side/java/src/test/java/com/testdroid/appium/ios/sample/IosAppiumExampleTest.java]: outdated

File: samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.ios.clientside.properties [hunk 6]
```diff
@@ -1,10 +1,9 @@
 platformName=iOS
-device=iphone
 deviceName=iOS Phone
 automationName=XCUITest
 bundleId=com.bitbar.testdroid.BitbarIOSSample
 bitbar_target=iOS
```
Comment (MaciejLendzion-smartbear) [inline samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.ios.clientside.properties:5]: `bitbar_target` is no longer used

File: samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.iosBiometry.clientside.properties [hunk 7]
```diff
@@ -1,11 +1,10 @@
 platformName=iOS
-device=iphone
 deviceName=iOS Phone
```
Comment (MaciejLendzion-smartbear) [inline samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.iosBiometry.clientside.properties:2]: `appium_deviceName`

File: samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.iosBiometry.clientside.properties [hunk 8]
```diff
@@ -1,11 +1,10 @@
...
 automationName=XCUITest
```
Comment (MaciejLendzion-smartbear) [inline samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.iosBiometry.clientside.properties:3]: `appium_automationName`

File: samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.androidBiometry.clientside.properties [hunk 9]
```diff
@@ -1,9 +1,10 @@
 platformName=Android
-deviceName=Android Device
 fullReset=false
 noReset=true
+automationName=uiautomator2
 bitbar_target=Android
```
Comment (MaciejLendzion-smartbear) [inline samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.androidBiometry.clientside.properties:5]: `bitbar_target` is no longer used

File: samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.ios.clientside.properties [hunk 10]
```diff
@@ -1,10 +1,9 @@
 platformName=iOS
-device=iphone
 deviceName=iOS Phone
```
Comment (MaciejLendzion-smartbear) [inline samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.ios.clientside.properties:2]: `appium_deviceName`

File: samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.iosBiometry.clientside.properties [hunk 11]
```diff
@@ -1,11 +1,10 @@
...
+bundleId=org.reactjs.native.example.BitBarSampleApp
 bitbar_target=iOS
```
Comment (MaciejLendzion-smartbear) [inline samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.iosBiometry.clientside.properties:5]: bitbar_target is no longer used

File: samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.androidBiometry.clientside.properties [hunk 12]
```diff
@@ -1,9 +1,10 @@
 platformName=Android
-deviceName=Android Device
 fullReset=false
```
Comment (MaciejLendzion-smartbear) [inline samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.androidBiometry.clientside.properties:2]: `appium_fullReset`

File: samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.androidBiometry.clientside.properties [hunk 13]
```diff
@@ -1,9 +1,10 @@
...
 noReset=true
```
Comment (MaciejLendzion-smartbear) [inline samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.androidBiometry.clientside.properties:3]: `appium_noReset`

File: samples/testing-frameworks/appium/client-side/java/src/test/java/com/testdroid/appium/BaseTest.java [hunk 14]
```diff
@@ -33,20 +33,6 @@ public void setUpAppiumDriver() throws IOException {
...
-            }
-            if (exportTestResultsToCloud()) {
-                LOGGER.debug("Exporting results enabled");
-                capabilities.setCapability("testdroid_junitWaitTime", 300);
-            }
-            capabilities.setCapability("testdroid_app", app);
-            capabilities.setCapability("testdroid_apiKey", getApiKey());
             LOGGER.debug("Setting client side specific capabilities... FINISHED");
         } else if (isServerSideTestRun()) {
             LOGGER.debug("Setting server side specific capabilities...");
```
[the remark this answers is not in the document]
Comment (eryk-tkaczewski) [reply samples/testing-frameworks/appium/client-side/java/src/test/java/com/testdroid/appium/BaseTest.java:38]: I moved it to getDesiredCapabilitiesFromProperties()

[the remark this answers is not in the document]
Comment (MaciejLendzion-smartbear) [reply samples/testing-frameworks/appium/client-side/java/src/test/java/com/testdroid/appium/BaseTest.java:38]: But the only thing that it does now is:
` LOGGER.debug("Setting client side specific capabilities...");`
`LOGGER.debug("Setting client side specific capabilities... FINISHED");`
It has no usage at all.

File: samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.androidBiometry.clientside.properties (hunk 12 above)
Comment (eryk-tkaczewski) [reply samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.androidBiometry.clientside.properties:2]: adds this in the getDesiredCapabilitiesFromProperties()

Comment (MaciejLendzion-smartbear) [reply samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.androidBiometry.clientside.properties:2]: Hm ok can you link where does it happen in code please?

File: samples/testing-frameworks/appium/client-side/java/src/test/java/com/testdroid/appium/BaseTest.java (hunk 14 above)
[the remark this answers is not in the document]
Comment (eryk-tkaczewski) [reply samples/testing-frameworks/appium/client-side/java/src/test/java/com/testdroid/appium/BaseTest.java]: true, so i moved all to getDesiredCapabilitiesFromProperties()

File: samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.androidBiometry.clientside.properties (hunk 12 above)
Comment (eryk-tkaczewski) [reply samples/testing-frameworks/appium/client-side/java/src/test/resources/desiredCapabilities.androidBiometry.clientside.properties]: never mind, my mistake, corrected

**Review by MaciejLendzion-smartbear — APPROVED** [review]

**Pull request merged**